test/jdk/javax/swing/JComboBox/ComboPopupBug.java

```diff
@@ -21,66 +21,72 @@
-            Click on Close and then click on JComboBox arrow button
-            to try to show combobox popup.
-            If IllegalStateException is thrown, test will automatically Fail
-            otherwise click Pass.""";
+    private static JFrame frame;
+    private static JButton closeButton;
+    private static JComboBox<String> comboBox;
+    private static Point comboBoxLocation;
+    private static Dimension comboBoxSize;
+    private static JRobot robot;
 
     public static void main(String[] args) throws Exception {
-        PassFailJFrame.builder()
-                .title("ComboPopup Instructions")
-                .instructions(instructionsText)
-                .testTimeOut(5)
-                .rows(10)
-                .columns(35)
-                .testUI(ComboPopupBug::createUI)
-                .build()
-                .awaitAndCheck();
-    }
+        robot = JRobot.getRobot();
+        SwingUtilities.invokeAndWait(() -> {
+            frame = new JFrame("ComboPopup");
 
-    private static JFrame createUI() {
-        JFrame frame = new JFrame("ComboPopup");
+            comboBox = new JComboBox<>();
+            comboBox.setEditable(true);
+            comboBox.addItem("test");
+            comboBox.addItem("test2");
+            comboBox.addItem("test3");
 
-        JComboBox<String> cb = new JComboBox<>();
-        cb.setEditable(true);
-        cb.addItem("test");
-        cb.addItem("test2");
-        cb.addItem("test3");
+            closeButton = new JButton("Close");
+            closeButton.addActionListener((e) -> {
+                        clickComboBox();
+                        frame.setVisible(false);
+                    });
 
-        JButton b = new JButton("Close");
-        b.addActionListener(
-                (e)->{
-                    try {
-                        Thread.sleep(3000);
-                    } catch (Exception ignored) {
-                    }
-                    frame.setVisible(false);
-                });
+            frame.getContentPane().add(comboBox, "North");
+            frame.getContentPane().add(closeButton, "South");
+            frame.setSize(200, 200);
+            frame.setVisible(true);
+        });
 
-        frame.getContentPane().add(cb, "North");
-        frame.getContentPane().add(b, "South");
-        frame.setSize(200, 200);
+        comboBoxLocation = comboBox.getLocationOnScreen();
+        comboBoxSize = comboBox.getSize();
+
+        try {
+            SwingUtilities.invokeAndWait(() -> closeButton.doClick());
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        finally {
+            SwingUtilities.invokeAndWait(() -> frame.dispose());
+        }
+    }
 
-        return frame;
+    public static void clickComboBox() {
+        int padding = 10;
+        robot.mouseMove(comboBoxLocation.x + comboBoxSize.width - padding, comboBoxLocation.y + comboBoxSize.height / 2);
```

Comment:
I'm sure this line doesn't fit into 80-column limit, I recommend wrapping it.
```suggestion
        robot.mouseMove(comboBoxLocation.x + comboBoxSize.width - padding,
                        comboBoxLocation.y + comboBoxSize.height / 2);
```

Padding or offset could be a constant declared statically.